R/functions-range.R

```diff
@@ -219,3 +220,173 @@ pl__datetime_ranges <- function(
+    step = 1,
+    ...,
+    dtype = pl$Int64) {
+  wrap({
+    check_dots_empty0(...)
+    if (is.null(end)) {
+      end <- start
+      start <- 0
+    }
+    int_ranges(
+      as_polars_expr(start)$`_rexpr`,
+      as_polars_expr(end)$`_rexpr`,
+      as_polars_expr(step)$`_rexpr`,
+      dtype$`_dt`
+    )
+  })
+}
+
+#' Generate a time range
+#'
+#' @inheritParams rlang::check_dots_empty0
+#' @inheritParams pl__datetime_range
+#' @param start Lower bound of the time range. If omitted, defaults to
+#' `00:00:00.000`.
+#' @param end Upper bound of the time range. If omitted, defaults to
+#' `23:59:59.999`
+#' @param interval Interval of the range periods, specified as a [difftime]
+#' or using the Polars duration string language (see details).
+#'
+#' @inheritSection polars_duration_string Polars duration string language
+#' @inherit as_polars_expr return
+#' @examplesIf requireNamespace("hms", quietly = TRUE)
+#' pl$select(
+#'   time = pl$time_range(
+#'     start = hms::parse_hms("14:00:00"),
+#'     interval = as.difftime("3:15:00")
+#'   )
+#' )
+pl__time_range <- function(
+    start = NULL,
+    end = NULL,
+    interval = "1h",
+    ...,
+    closed = "both") {
+  wrap({
+    check_installed("hms")
+    check_dots_empty0(...)
+    closed <- arg_match0(closed, values = c("both", "left", "none", "right"))
+    interval <- parse_as_polars_duration_string(interval)
+    for (unit in c("y", "mo", "w", "d")) {
+      if (grepl(unit, interval)) {
+        abort(paste0("invalid unit in `interval`: found \"", unit, "\""))
+      }
+    }
+    if (is.null(start)) {
+      start <- hms::parse_hms("00:00:00")
+    }
+    if (is.null(end)) {
+      end <- hms::parse_hms("23:59:59.999")
+    }
+    time_range(
+      as_polars_expr(start)$`_rexpr`,
+      as_polars_expr(end)$`_rexpr`,
+      interval,
+      closed
+    )
+  })
+}
+
+#' Create a column of time ranges
+#'
+#' @inheritParams rlang::check_dots_empty0
+#' @inheritParams pl__time_range
+#'
+#' @inheritSection polars_duration_string Polars duration string language
+#' @inherit as_polars_expr return
+#' @examplesIf requireNamespace("hms", quietly = TRUE)
+#' df <- pl$DataFrame(
+#'   start = hms::parse_hms(c("09:00:00", "10:00:00")),
+#'   end = hms::parse_hms(c("11:00:00", "11:00:00"))
+#' )
+#' df$with_columns(time_range = pl$time_ranges("start", "end"))
+pl__time_ranges <- function(
+    start = NULL,
+    end = NULL,
+    interval = "1h",
+    ...,
+    closed = "both") {
+  wrap({
+    check_installed("hms")
+    check_dots_empty0(...)
+    closed <- arg_match0(closed, values = c("both", "left", "none", "right"))
+    interval <- parse_as_polars_duration_string(interval)
+    for (unit in c("y", "mo", "w", "d")) {
+      if (grepl(unit, interval)) {
+        abort(paste0("invalid interval unit: found ", unit))
+      }
+    }
+    if (is.null(start)) {
+      start <- hms::parse_hms("00:00:00")
+    }
+    if (is.null(end)) {
+      end <- hms::parse_hms("23:59:59.999")
```

Comment:
If you do this, it will not work if hms is not installed. Cast literal from Int32 to Time.

Reply:
Forgot how to create time columns without hms, thanks